src/Validation/Validator.php

```diff
@@ -314,14 +321,40 @@ public function add($field, $name, $rule = [])
             $rules = $name;
         }
 
+        if ($this->_autoRequire) {
+            $rules += [
+                '_requirePresence' => []
+            ];
+
+            $rules['_requirePresence'] += [
+                'mode' => true,
+                'message' => null
+            ];
+
+            $this->requirePresence($field, $rules['_requirePresence']['mode'], $rules['_requirePresence']['message']);
+        }
+        unset($rules['_requirePresence']);
+
         foreach ($rules as $name => $rule) {
-            $field->add($name, $rule);
+            $ruleSet->add($name, $rule);
         }
 
         return $this;
     }
 
     /**
+     * Adds automaticly requirePresence on each field added through add method
+     *
+     * @param bool $require whether auto require is on or off
+     * @return $this
+     */
+    public function autoRequire($require)
```

Comment:
I'm not sure about this. It could make validators harder to work with. The position of this method in a method chain changes the behavior of method calls around it. For example.

```php
$validator->
  ->email('email')
  ->autoRequire()
  ->minLength('name', 22);
```

and

```php
$validator->
  ->autoRequire()
  ->email('email')
  ->minLength('name', 22);
```

Do very different things.